app/src/main/kotlin/github/com/rhacco/dota2androidapp/viewmodel/MatchesViewModel.kt

```diff
@@ -0,0 +1,80 @@
+package github.com.rhacco.dota2androidapp.viewmodel
+
+import android.app.Application
+import android.arch.lifecycle.AndroidViewModel
+import android.arch.lifecycle.MediatorLiveData
+import android.util.Log
+import github.com.rhacco.dota2androidapp.App
+import github.com.rhacco.dota2androidapp.R
+import github.com.rhacco.dota2androidapp.api.TopLiveGamesResponse
+import github.com.rhacco.dota2androidapp.lists.LiveMatchesItemData
+import github.com.rhacco.dota2androidapp.sources.repos.matches.RealtimeStatsRepository
+import github.com.rhacco.dota2androidapp.sources.repos.matches.TopLiveGamesRepository
+import io.reactivex.disposables.CompositeDisposable
+
+class MatchesViewModel(application: Application) : AndroidViewModel(application) {
+    private val mIsLoading = MediatorLiveData<Boolean>()
+    private val mDisposables = CompositeDisposable()
+    val mLiveMatchesQuery = MediatorLiveData<List<TopLiveGamesResponse.Game>>()
+    val mLiveMatchesItemDataQuery = MediatorLiveData<LiveMatchesItemData>()
+
+    override fun onCleared() = mDisposables.clear()
+
+    fun getLiveMatches() {
+        mIsLoading.value = true
+        mDisposables.add(TopLiveGamesRepository.getTopLiveGames()
+                .subscribe(
+                        { result ->
+                            mIsLoading.value = false
+                            mLiveMatchesQuery.value = result
+                        },
+                        { error ->
+                            mIsLoading.value = false
+                            Log.d(App.instance.getString(R.string.log_msg_debug),
+                                    "Failed to fetch top live games: " + error)
+                        }
+                ))
+    }
+
+    fun getLiveMatchesItemData(averageMMR: Int, serverSteamId: Long) {
+        mIsLoading.value = true
+        mDisposables.add(RealtimeStatsRepository.getRealtimeStats(serverSteamId)
+                .subscribe(
+                        { result ->
+                            mIsLoading.value = false
+                            val newItemData = LiveMatchesItemData()
+                            newItemData.mAverageMMR = averageMMR
+                            if (averageMMR < 1)
+                                newItemData.mTitle = "Tournament Match (Match ID " +
```

Comment:
Make Tournament Match and Ranked Match be strings in the strings.xml that take an argument.

Reply:
Done